src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/GitDiffedCodeExtractionManager.java

```diff
@@ -41,308 +40,298 @@
-		changedFile);
+		Map<Edit, List<KnowledgeElement>> elementsByEditOld = getRationaleFromComments(false, commentsInFileA,
+				changedFile);
 
-	Map<Edit, List<KnowledgeElement>> elementsByEditOld = getRationaleFromComments(false, commentsInFileA,
-		changedFile);
+		returnCodeExtractionResult.diffedElementsInNewerVersion = elementsByEditNew;
+		returnCodeExtractionResult.diffedElementsInOlderVersion = elementsByEditOld;
 
-	returnCodeExtractionResult.diffedElementsInNewerVersion = elementsByEditNew;
-	returnCodeExtractionResult.diffedElementsInOlderVersion = elementsByEditOld;
+		return returnCodeExtractionResult;
+	}
 
-	return returnCodeExtractionResult;
-    }
+	private Map<Edit, List<KnowledgeElement>> getRationaleFromComments(boolean newerFile,
+			List<CodeComment> commentsInFile, ChangedFile changedFile) {
 
-    private Map<Edit, List<KnowledgeElement>> getRationaleFromComments(boolean newerFile,
-	    List<CodeComment> commentsInFile, ChangedFile changedFile) {
+		Map<Edit, List<KnowledgeElement>> returnMap = new HashMap<>();
 
-	Map<Edit, List<KnowledgeElement>> returnMap = new HashMap<>();
+		RationaleFromDiffCodeCommentExtractor rationaleFromDiffCodeCommentExtractor = new RationaleFromDiffCodeCommentExtractor(
+				commentsInFile, changedFile.getEditList());
 
-	RationaleFromDiffCodeCommentExtractor rationaleFromDiffCodeCommentExtractor = new RationaleFromDiffCodeCommentExtractor(
-		commentsInFile, changedFile.getEditList());
+		while (rationaleFromDiffCodeCommentExtractor.next()) {
+			returnMap.putAll(rationaleFromDiffCodeCommentExtractor.getRationaleFromComment(newerFile, returnMap));
+		}
 
-	while (rationaleFromDiffCodeCommentExtractor.next()) {
-	    returnMap.putAll(rationaleFromDiffCodeCommentExtractor.getRationaleFromComment(newerFile, returnMap));
+		return returnMap;
 	}
 
-	return returnMap;
-    }
-
-    private List<CodeComment> getCommentsFromFile(String inspectedFileRelativePath, boolean fromNewerFile) {
-	File resultingFile = getInspectedFileAbsolutePath(inspectedFileRelativePath, fromNewerFile);
-	if (!resultingFile.isFile()) {
-	    // LOGGER.error("Expected file " + resultingFile.getAbsolutePath() + " to
-	    // exist!");
-	    return null;
-	} else if (!resultingFile.canRead()) {
-	    // LOGGER.error("Expected file " + resultingFile.getAbsolutePath() + " to be
-	    // readable!");
-	    return null;
-	} else {
-	    CodeCommentParser commentParser = getCodeCommentParser(inspectedFileRelativePath);
-	    if (commentParser == null) {
-		LOGGER.info("No parser could be found for file " + resultingFile.getName());
-		return null;
-	    } else {
-		return commentParser.getComments(resultingFile);
-	    }
+	private List<CodeComment> getCommentsFromFile(String inspectedFileRelativePath, boolean fromNewerFile) {
+		File resultingFile = getInspectedFileAbsolutePath(inspectedFileRelativePath, fromNewerFile);
+		if (!resultingFile.isFile()) {
```

Comment:
Das IF-Statement kann stark vereinfacht werden da nur in einem Fall etwas zurück geliefert wird.